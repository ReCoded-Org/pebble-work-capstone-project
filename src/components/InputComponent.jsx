const InputComponent = ({
    placeholder = "Your input here",
    type = "text",
    callback,
    className = "h-full w-full rounded-md border-2 border-black p-2",
    label,
    name,
    id,
    value,
}) => {
    return (
        <>
            <input
                className={className}
                type={type}
                placeholder={placeholder}
                onChange={callback}
                name={name}
                id={id}
                value={value}
            />
            <label htmlFor={name}>{label}</label>
        </>
    );
};

export default InputComponent;
