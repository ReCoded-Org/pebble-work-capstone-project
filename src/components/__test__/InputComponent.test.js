import renderer from "react-test-renderer";

import InputComponent from "../InputComponent";

it("renders correctly", () => {
    const tree = renderer.create(<InputComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
