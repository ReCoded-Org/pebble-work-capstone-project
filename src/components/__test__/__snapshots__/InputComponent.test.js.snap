// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <input
    className="h-full w-full rounded-md border-2 border-black p-2"
    placeholder="Your input here"
    type="text"
  />,
  <label />,
]
`;
